### PR TITLE
swapped out Wire references for Hyperion package

### PR DESCRIPTION
**Pull request by Aaronontheweb**
This PR swaps out the underlying Wire dependencies for Hyperion, in order to stop our users from getting GPL'd by upgrading to a new version of Wire.

I propose doing this across two releases

1. Release the current Akka.Serailization.Wire, complete with Hyperion dependencies, to move net new users onto the latest Hyperion version.
2. Deprecate the Akka.Serialization.Wire package and replace it with a Akka.Serialization.Hyperion package and update the necessary documentation in a subsequent release.

Any thoughts on the above?

### CODE REVIEW

File: src/Akka.sln [hunk 1]
```diff
@@ -186,7 +186,7 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Akka.Serialization.Wire", "
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Akka.Serialization.TestKit", "contrib\serializers\Akka.Serialization.TestKit\Akka.Serialization.TestKit.csproj", "{CAA97041-CFC0-4081-9BD2-8B139E62A611}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Akka.Serialization.WireTests", "contrib\serializers\Akka.Serialization.WireTests\Akka.Serialization.WireTests.csproj", "{402FA351-D6C6-40FD-8868-F07156035919}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Akka.Serialization.Wire.Tests", "contrib\serializers\Akka.Serialization.WireTests\Akka.Serialization.Wire.Tests.csproj", "{402FA351-D6C6-40FD-8868-F07156035919}"
```
Comment (Aaronontheweb) [inline src/Akka.sln]: Renamed the project so it would get picked up by our FAKE script for serialization purposes.

**Comment by sean-gilliam** [comment]
Sounds good to me

**Comment by Aaronontheweb** [comment]
cc @Horusiath looks like we have an issue with the Akka.DistributedData bindings and the Hyperion serializer. Anything special that needs to be done with Hyperion or its serializer definition here? Take a look at the unit test failures. I can get them to fail locally as well.

**Comment by Horusiath** [comment]
@Aaronontheweb I think this is a result of changes made between Wire v0.7.1 and v0.8.1 - existing Akka.Serialization.Wire references the first, while Hyperion is based on the second.

It's one of the edge cases, we didn't have a test for - situation when immutable collection is referenced through interface (in this case `IImmutableDictionary<,>`) as a field in some custom class. I'll try to fix it today.

**Comment by Aaronontheweb** [comment]
@Horusiath thanks, we'll update the Hyperion package today

**Comment by Aaronontheweb** [comment]
Resolves https://github.com/akkadotnet/akka.net/issues/2435 as well

**Comment by Horusiath** [comment]
Failures in ddata are caused by the problem I've solved in https://github.com/akkadotnet/Hyperion/pull/13

**Comment by Aaronontheweb** [comment]
@Horusiath yep, that's all in there. Newest version of Hyperion is going up on NuGet now.

**Comment by Aaronontheweb** [comment]
Upgraded to 0.9.1

**Comment by Aaronontheweb** [comment]
@Horusiath still have 7 spec failures on the DistributedData stuff, down from 9. http://petabridge-ci.cloudapp.net/viewLog.html?buildId=18997&buildTypeId=AkkaNet_AkkaNetLinuxMonoUnitTests&guest=1

**Comment by Horusiath** [comment]
@Aaronontheweb I've pulled your PR and tracked down the issue. The problem lies [here](https://github.com/akkadotnet/Hyperion/blob/dev/Hyperion/SerializerFactories/EnumerableSerializerFactory.cs#L98). 

To deserialize enumerables (which ddata collections actually are), existing serializer was trying to find either `AddRange` or `Add` method and invoke it. Problem is that it automatically assume, that signature of add method follows the rules:

- `void Add(T item)`
- `void AddRange(IEnumerable<T> items)`

...which is not true in case of ddata i.e. `GSet<>.Add` returns a whole new GSet instance (that's why test fails on assertion), while `LWWDictionary<>.Add` has different input params (that's why we have reflection exception).

**Comment by Aaronontheweb** [comment]
Found the issue:

```
[Test collection for Akka.Serialization.Wire.Tests.WireTests (1)] Akka.Serialization.Wire.Tests.WireTests.CanSerializeProps (4s)
[16:37:30]W:		 [docker] Stacktrace:
[16:37:30]W:		 [docker] 
[16:37:30]W:		 [docker]   at <unknown> <0xffffffff>
[16:37:30]W:		 [docker]   at (wrapper managed-to-native) System.RuntimeTypeHandle.type_is_assignable_from (System.Type,System.Type) <0x00057>
[16:37:30]W:		 [docker]   at System.RuntimeTypeHandle.CanCastTo (System.RuntimeType,System.RuntimeType) <0x00020>
[16:37:30]W:		 [docker]   at System.RuntimeType.IsAssignableFrom (System.Type) <0x000b6>
[16:37:30]W:		 [docker]   at Akka.Actor.Props.CreateProducer (System.Type,object[]) <0x0005d>
[16:37:30]W:		 [docker]   at Akka.Actor.Props..ctor (Akka.Actor.Deploy,System.Type,object[]) <0x00087>
[16:37:30]W:		 [docker]   at Akka.Actor.Props/PropsSurrogate.FromSurrogate (Akka.Actor.ActorSystem) <0x0005b>
[16:37:30]W:		 [docker]   at Akka.Serialization.WireSerializer/<WireSerializer>c__AnonStorey0.<>m__1 (Akka.Util.ISurrogate) <0x00025>
[16:37:30]W:		 [docker]   at Hyperion.Surrogate`2/<>c__DisplayClass0_0<TSource_REF, TSurrogate_REF>.<.ctor>b__1 (object) <0x00053>
[16:37:30]W:		 [docker]   at Hyperion.ValueSerializers.FromSurrogateSerializer.ReadValue (System.IO.Stream,Hyperion.DeserializerSession) <0x0003d>
[16:37:30]W:		 [docker]   at Hyperion.Serializer.Deserialize<T_REF> (System.IO.Stream) <0x00044>
[16:37:30]W:		 [docker]   at Akka.Serialization.WireSerializer.FromBinary (byte[],System.Type) <0x00063>
[16:37:30]W:		 [docker]   at Akka.Tests.Serialization.AkkaSerializationSpec.AssertEqual<T_REF> (T_REF) <0x0006d>
[16:37:30]W:		 [docker]   at Akka.Tests.Serialization.AkkaSerializationSpec.CanSerializeProps () <0x00077>
[16:37:30]W:		 [docker]   at (wrapper runtime-invoke) object.runtime_invoke_void__this__ (object,intptr,intptr,intptr) <0x000c8>
[16:37:30]W:		 [docker]   at <unknown> <0xffffffff>
[16:37:30]W:		 [docker]   at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&) <0x0006c>
[16:37:30]W:		 [docker]   at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) <0x000b7>
[16:37:30]W:		 [docker]   at System.Reflection.MethodBase.Invoke (object,object[]) <0x0002a>
[16:37:30]W:		 [docker]   at Xunit.Sdk.TestInvoker`1<TTestCase_REF>.CallTestMethod (object) <0x00043>
```

Looks like there's something going on in that assertion specifically that triggers a native error in the Mono runtime. Working on resolving some Mono build issues with Hyperion itself so I can get a better look.

**Comment by Aaronontheweb** [comment]
Trying this out with a Mono 4.6.1 build

**Comment by Aaronontheweb** [comment]
Looks like the Mono upgrade fixed the serialization issue.

**Comment by Horusiath** [comment]
Is it ready to merge?

**Comment by Aaronontheweb** [comment]
Yep
